src/views/Map/AgentMAp.js

```diff
@@ -0,0 +1,173 @@
+  }
+
+  initMap() {
+    let google = window.google;
+    let addRoute = this.addRoute;
+    const handleLocationError = this.handleLocationError;
+    let processSVData = this.processSVData;
+
+    let latlong = { lat: 6.50954222424663, lng: 3.370920970651687 };
+    let sv = new google.maps.StreetViewService();
+    panorama = new google.maps.StreetViewPanorama(document.getElementById('pano'));
+    // Set up the map.
+    map = new google.maps.Map(document.getElementById('map'), {
+      center: latlong,
+      zoom: 15,
+      streetViewControl: false
+    });
+
+    infoWindow = new google.maps.InfoWindow();
+    // Try HTML5 geolocation.
+    if (navigator.geolocation) {
+      pos = { lat: 6.60954222424663, lng: 3.370920970651687 };
+      navigator.geolocation.getCurrentPosition((position) => {
+        pos = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        };
+        if (pos !== undefined) {
+          console.log(pos);
+          addRoute();
+        }
+      }, () => {
+        addRoute();
+        handleLocationError(true, infoWindow, map.getCenter());
+      });
+    } else {
+      addRoute();
+      // Browser doesn't support Geolocation
+      handleLocationError(false, infoWindow, map.getCenter());
+    }
+    // Set the initial Street View camera to the center of the map
+    sv.getPanorama({ location: null, radius: 50 }, processSVData);
+    map.addListener('click', (event) => {
+      sv.getPanorama({ location: event.latLng, radius: 50 }, processSVData);
+    });
+
+  }
+
+  addRoute() {
+    let google = window.google;
+    let start;
+    let directionsRenderer = new google.maps.DirectionsRenderer();
+    let directionsService = new google.maps.DirectionsService();
+    directionsRenderer.setMap(map);
+    directionsRenderer.setPanel(document.getElementById('right-panel'));
+    if (pos === undefined) {
+      start = { lat: 6.60157422424999, lng: 3.360932070651222 };
+    } else {
+      start = pos;
+    }
+    let end = { lat: 6.60157222424999, lng: 3.360920570651222 };
+    directionsService.route({
+      origin: start,
+      destination: end,
+      travelMode: 'DRIVING'
+    }, (response, status) => {
+      if (status === 'OK') {
+        directionsRenderer.setDirections(response);
+      } else {
+        window.alert('Directions request failed due to ' + status);
+      }
+    });
+  }
+
+  processSVData(data, status) {
+    if (status === 'OK') {
+      if (data !== null) {
+        let google = window.google;
+
+        let marker = new google.maps.Marker({
+          position: data.location.latLng,
+          map: map,
+          title: data.location.description
+        });
+      }
+
+      panorama.setPano(data.location.pano);
+      panorama.setPov({
+        heading: 270,
+        pitch: 0
+      });
+      panorama.setVisible(true);
+
+      marker.addListener('click', () => {
+        let markerPanoID = data.location.pano;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['markerPanoID' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/BuildForSDG/Team-064-Frontend/pullRequest?prid=5703258)